tests/test_gdist.py

```diff
@@ -138,3 +140,25 @@ def test_hedgehog_mesh(self):
         # test if the obtained matrix is symmetric
         assert (abs(distances - distances.T) > epsilon).nnz == 0
         assert np.max(distances) <= 1.45
+
+
+class TestDistanceMatrixOfSelectedPoints:
+    def test_flat_triangular_mesh(self):
+        data = np.loadtxt("data/flat_triangular_mesh.txt", skiprows=1)
+        vertices = data[0:121].astype(np.float64)
+        triangles = data[121:].astype(np.int32)
+        points = np.array([2, 5, 10], dtype=np.int32)
+        distances = gdist.distance_matrix_of_selected_points(
+            vertices,
+            triangles,
+            points,
+        )
+        epsilon = 1e-6
+        expected = np.loadtxt("data/flat_triangular_mesh_pairwise_matrix.txt")
+        np.testing.assert_array_almost_equal(distances.toarray(), expected)
+        # test if the obtained matrix is symmetric
+        assert (abs(distances - distances.T) > epsilon).nnz == 0
+        # make sure number of non-zero elements are twice of number of points
+        assert distances.nnz == 2 * points.shape[0]
```

Comment:
Does this assert 2*pojnts no scale?
Shouldn't it be N^2 - N (-N for the diagonal) ?

Reply:
Yes, you are right. Thanks!

Reply:
Fixed it.